### ISSUE TITLE
tf.ready() crashing React Native (Expo managed) app with no error

### ISSUE THREAD

**Issue by darringrove** [tensorflow/tfjs#6710]
This was working fine yesterday, but started crashing today.
No changes to package.json.
Only known upgrade was Expo CLI upgrade to 6.0.1 (early than today).
No other known new software installed between yesterday and today.

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js): 
No. Example code is crashing also.

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Started with macOS Monterey 12.4, upgraded to 12.5 with same result

- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
iPhone 12 Pro, iOS 15.6

- TensorFlow.js installed from (npm or script link):
yarn

- TensorFlow.js version (use command below):
From the pose-detection example

```
    "@mediapipe/pose": "~0.4.0",
    "@react-native-async-storage/async-storage": "~1.17.3",
    "@tensorflow-models/pose-detection": "2.0.0",
    "@tensorflow/tfjs": "3.18.0",
    "@tensorflow/tfjs-react-native": "0.8.0",
    "expo": "~45.0.6",
    "expo-camera": "^12.2.0",
    "expo-file-system": "^14.0.0",
    "expo-gl": "^11.3.0",
    "expo-gl-cpp": "^11.3.0",
    "expo-screen-orientation": "~4.2.0",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-native": "0.68.2",
    "react-native-fs": "2.14.1",
    "react-native-gesture-handler": "~2.2.1",
    "react-native-svg": "12.3.0"
```

- Browser version: n/a
- Tensorflow.js Converter Version: n/a

**Describe the current behavior**
App crashing without error on the call to await tf.ready();

**Describe the expected behavior**
await tf.ready() is successful

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/CodePen/any notebook.

[](https://github.com/tensorflow/tfjs-examples/tree/master/react-native/pose-detection)

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by rthadur** [comment]
@darringrove sorry for the delay in the original issue https://github.com/tensorflow/tfjs/issues/6611 , I have escalated it to Jing. Closing this duplicate issue. Thank you

**Comment by darringrove** [comment]
@rthadur This is a new issue. [#6611](https://github.com/tensorflow/tfjs/issues/6611) is an issue with a bare React Native app in createDetector. This issue is with an Expo managed app which is failing in tf.ready().

**Comment by darringrove** [comment]
@jinjingforever I took a couple of additional steps to diagnose this.

1. Created a new expo managed app from the typescript template
2. Added the packages specified in tfjs-react-native
3. Added useEffect code similar to the pose-detection example to call tf.ready()

Just that much crashes the app.
I attempted to upgrade packages to see if that made a difference but it did not.
It's still a mystery what changed last night, since this worked fine yesterday.

I'm interested to know if you can reproduce this with your pose-detection example.

Thanks in advance.

**Comment by darringrove** [comment]
@rthadur @jinjingforever I'm not sure if this is a clue or not. I noticed in some other threads that calling `tf.setBackend` solved crashing problems.

I tried `console.log(tf.getBackend());` before `await tf.ready()`. It returns `undefined`.

I tried `await tf.setBackend('cpu');` before `await tf.ready()` (not sure this is the right backend), but with this `tf.ready()` returned, but then `await posedetection.createDetector` crashed the app.

**Comment by darringrove** [comment]
@jinjingforever Any update on this? Our development efforts, which were moving forward nicely, are now completely blocked by this crashing issue.

**Comment by jinjingforever** [comment]
Hi @darringrove 

Sorry, I was busy with some other projects...

I took a look, and it is probably caused by [this bug](https://github.com/expo/expo/issues/18493). Apparently, expo go app was updated 3 days ago, which added the support to expo sdk 46. But for some reason, it crashes apps that uses expo-gl and expo sdk 45. I updated pose-detection to expo sdk 46 and it seems to work now ([PR](https://github.com/tensorflow/tfjs-examples/pull/868)). Please give it a try. Thanks!

**Comment by darringrove** [comment]
@jinjingforever That was it! The pose-detection example is working again. (Now to upgrade my app and settle other package updates.) Thanks much!!

**Comment by Rhysjc** [comment]
@jinjingforever This is occurring again on Expo 47

**Comment by fbotalla** [comment]
tf.ready() is crashing the app on Expo 47 for both managed and bare projects.
